frontend/javascripts/admin/team/edit_team_modal_view.tsx

```diff
@@ -15,27 +15,31 @@ type Props = {
 
 function EditTeamModalForm({ onCancel, isOpen, team }: Props) {
   const [autoCompleteValue, setAutoCompleteValue] = useState("");
+  const [dropDownVisible, setDropDownVisible] = useState(false);
   const onChange = (newValue: string) => setAutoCompleteValue(newValue);
   const [users, setUsers] = useState<APIUser[] | null>(null);
-  const fetchUsers = async () => setTimeout(async () => setUsers(await getEditableUsers()), 5000); // TODO delete; only testing purposes
-  //const fetchUsers = async () => setUsers(await getEditableUsers());
+  const [pendingRequestsCounter, setPendingRequestsCounter] = useState(0);
+  const fetchUsers = async () => setTimeout(async () => setUsers(await getEditableUsers()), 2000); //TODO
   useEffect(() => {
     fetchUsers();
   }, []);
 
   if (team === null) return null;
   const updateTeamMembership = async (user: APIUser, newTeams: APITeamMembership[]) => {
+    setPendingRequestsCounter(1);
```

Comment:
> Initially I tried that, but it didn't work. The counter ended up being smaller than zero at the beginning of the second request, although it was zero after the first request.

Just fyi: incrementing/decrementing is a classic example for using a functional update. See the docs [here](https://legacy.reactjs.org/docs/hooks-reference.html#functional-updates). When not doing so, it can easily happen that the calculation is done on "outdated" values. This can happen because the callback that is executed is still seeing old values in its scope. The scope will only update on the next re-render, but an event handler can be kicked off by the re-rendering happens.

> Alright, I'll do that!

great :+1: the functional update won't be necessary then.

Reply:
Very interesting, thank you for the hint!